File: samples/generators/angular/src/appgenerated/model/securite/utilisateur/utilisateur-item.ts
////
//// ATTENTION CE FICHIER EST GENERE AUTOMATIQUEMENT !
////

import {DO_CODE, DO_EMAIL, DO_ID, DO_LIBELLE} from "@domains";

import {TypeUtilisateurCode} from "./references";

export interface UtilisateurItem {
    id?: number,
    nom?: string,
    prenom?: string,
    email?: string,
    typeUtilisateurCode?: TypeUtilisateurCode
}

export const UtilisateurItemEntity = {
    id: {
        type: "field",
        name: "id",
        domain: DO_ID,
        isRequired: false,
        label: "securite.utilisateur.utilisateur.id"
    },
    nom: {
        type: "field",
        name: "nom",
        domain: DO_LIBELLE,
        isRequired: true,
        label: "securite.utilisateur.utilisateur.nom"
    },
    prenom: {
        type: "field",
        name: "prenom",
        domain: DO_LIBELLE,
        isRequired: true,
        label: "securite.utilisateur.utilisateur.prenom"
    },
    email: {
        type: "field",
        name: "email",
        domain: DO_EMAIL,
        isRequired: true,
        label: "securite.utilisateur.utilisateur.email"
    },
    typeUtilisateurCode: {
        type: "field",
        name: "typeUtilisateurCode",
        domain: DO_CODE,
        isRequired: true,
        defaultValue: "GEST",
        label: "securite.utilisateur.utilisateur.typeUtilisateurCode"
    }
} as const

File: samples/generators/jpa/src/main/javagen/topmodel/jpa/sample/demo/dtos/securite/utilisateur/UtilisateurItem.java
////
//// ATTENTION CE FICHIER EST GENERE AUTOMATIQUEMENT !
////

package topmodel.jpa.sample.demo.dtos.securite.utilisateur;

import jakarta.annotation.Generated;

import topmodel.jpa.sample.demo.enums.securite.utilisateur.TypeUtilisateurCode;

/**
 * Détail d'un utilisateur en liste.
 */
@Generated("TopModel : https://github.com/klee-contrib/topmodel")
public interface UtilisateurItem {

	/**
	 * Getter for id.
	 */
	Integer getId();

	/**
	 * Getter for nom.
	 */
	String getNom();

	/**
	 * Getter for prenom.
	 */
	String getPrenom();

	/**
	 * Getter for email.
	 */
	String getEmail();

	/**
	 * Getter for typeUtilisateurCode.
	 */
	TypeUtilisateurCode getTypeUtilisateurCode();

	/**
	 * Hydrate values of instance.
	 * @param id value to set.
	 * @param nom value to set.
	 * @param prenom value to set.
	 * @param email value to set.
	 * @param typeUtilisateurCode value to set.
	 */
	void hydrate(Integer id, String nom, String prenom, String email, TypeUtilisateurCode typeUtilisateurCode);
}
